data/api/client-server/report_content.yaml

```
@@ -29,11 +29,14 @@ securityDefinitions:
 paths:
   "/rooms/{roomId}/report/{eventId}":
     post:
-      summary: Reports an event as inappropriate.
+      summary: Reports an event as inappropriate. You must have permission to
+        retrieve this event e.g. by being a member in the room for this event.
       description: |-
         Reports an event as inappropriate to the server, which may then notify
         the appropriate people.
```

Could you add a line about combating timing attacks by adding artificial delay to the response, as mentioned in [this bit of the MSC](https://github.com/matrix-org/matrix-spec-proposals/blob/hs/reporting-fixes/proposals/2249-report-require-joined.md#security-considerations)? Something like:

```suggestion
        Reports an event as inappropriate to the server, which may then notify
        the appropriate people.
        
        It may be possible for clients to deduce whether an event exists by
        timing the response, as only a report for an event that does exist
        will require the homeserver to check whether a user is joined to
        the room. To combat this, homeserver implementations may add
        a random delay when generating a response.
```

---

without "may":


```suggestion
        Reports an event as inappropriate to the server, which may then notify
        the appropriate people.
        
        It can be possible for clients to deduce whether an event exists by
        timing the response, as only a report for an event that does exist
        will require the homeserver to check whether a user is joined to
        the room. To combat this, homeserver implementations are encouraged
        to add a random delay when generating the response.
```